### PR TITLE
Add new android permissions to be able to change the wifi status

### PR DESCRIPTION
**Pull request by yoprogramo**
If you try to change existing wifi connection with a new one you'll need this permissions.

### CODE REVIEW

File: plugin.xml [hunk 1]
```diff
@@ -20,6 +20,10 @@
         <config-file target="AndroidManifest.xml" parent="/*">
             <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
             <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
+            <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
+            <uses-permission android:name="android.permission.INTERNET" />
+            <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+            <uses-permission android:name="android.permission.OVERRIDE_CONFIG_WIFI" />
```
Comment (sushichop) [inline plugin.xml:26]: Thanks for your pull request.

I'm not sure for now, but it seems that the minimum requirement is just to add `android.permission.ACCESS_FINE_LOCATION` according to the following document up to API level 28.
In addition, I need to use new network API to support the applications targeting API level 29(Android 10)...

Do you have any idea?

https://developer.android.com/guide/topics/connectivity/wifi-scan
https://developer.android.com/reference/android/net/wifi/WifiManager#addNetwork(android.net.wifi.WifiConfiguration)

**Review by sushichop — COMMENTED** [review]
@yoprogramo 
Thanks for your pull request! 
I left a comment. Could you check it out?

File: plugin.xml (hunk 1 above)
Comment (yoprogramo) [reply plugin.xml:26]: Hello,

I had to use this permissions in one of our apps (nomorepass.com) that is tested and in use with Android 10 mobiles. But I target Android 28 in the compilation. It is a cordova application, so, I'm not sure about how to support directly the new network API.

Comment (sushichop) [reply plugin.xml:26]: Thanks for sharing your idea!
I'll look into it this weekend.

Comment (sushichop) [reply plugin.xml:26]: I investigated using the latest environment(cordova@9.0.0, cordova-android@9.0.0, and Android 10 device). I confirmed the following.
- Any additional permission was not needed to connect Wi-Fi AP. 
- The device did not connect to Wi-Fi AP only when the target SDK version is 29(probably or later).
- `android.permission.INTERNET` is automatically added by cordova-android.

So I think the permissions you suggested are not directly related to connect Wi-Fi AP, on the other hand you application still needs it🙂

Then, I haven't decided yet whether I support the target SDK version 29 using new API.

Thanks!

**Comment by sushichop** [comment]
/danger to check files with latest files

**Comment by sushichop** [comment]
/danger to check files with the latest files

**Comment by sushichop** [comment]
/danger to check files with the latest files

**Comment by sushichop** [comment]
I have released [cordova-plugin-wifi-manager@0.3.0](https://www.npmjs.com/package/cordova-plugin-wifi-manager) which  supports Android 10 or later. 
Try to use it if necessary.

Thanks

**Pull request closed without merging**